debian/rules

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/make -f
+# See debhelper(7) (uncomment to enable)
+# output every command that modifies files on the build system.
+#export DH_VERBOSE = 1
+
+
+# see FEATURE AREAS in dpkg-buildflags(1)
+export DEB_BUILD_MAINT_OPTIONS = hardening=+all
+
+# see ENVIRONMENT in dpkg-buildflags(1)
+# package maintainers to append CFLAGS
+#export DEB_CFLAGS_MAINT_APPEND  = -Wall -pedantic
+# package maintainers to append LDFLAGS
+#export DEB_LDFLAGS_MAINT_APPEND = -Wl,--as-needed
+
+
+%:
+	dh $@ --buildsystem=cmake
+
+
+# dh_make generated override targets
+# This is example for Cmake (See https://bugs.debian.org/641051 )
+override_dh_auto_configure:
+	dh_auto_configure -- -DCMAKE_BUILD_TYPE=Release -DARCH=$(DEB_TARGET_ARCH)
```

Comment:
You're leaving dependencies on autodetect, but not specifying them in depends.
AFAIU, this could be using system-wide or vendored dependencies depending on whether they happen to be installed when running the build. I think we should instead specify `-DWITH_OWN_<dependency>` explicitly for all the deps (either to yes or no, but I'm thinking YES will be better here, as we need a specific seccomp, and I don't know if the other deps are even packaged for Debian).